src/Yahoo/Provider.php

```diff
@@ -57,11 +60,11 @@ protected function getUserByToken($token)
     protected function mapUserToObject(array $user)
     {
         return (new User())->setRaw($user)->map([
-            'id'       => $user['guid'],
-            'nickname' => $user['nickname'],
-            'name'     => trim(sprintf('%s %s', Arr::get($user, 'givenName'), Arr::get($user, 'familyName'))),
-            'email'    => Arr::get($user, 'emails.0.handle'),
-            'avatar'   => Arr::get($user, 'image.imageUrl'),
+            'id'       => $user['sub'],
+            'nickname' => $user['nickname'] || $user['sub'],
```

Comment:
```suggestion
            'nickname' => Arr::get($user, 'nickname', Arr::get($user, 'sub')),
```

Reply:
done

Reply:
Thanks for this suggestion, safer way to get data from array